Dfe.Academies.Academisation.Service/Commands/Application/ApplicationSubmitCommand.cs

```diff
@@ -31,36 +36,37 @@ public async Task<CommandResult> Execute(int applicationId)
 				return new NotFoundCommandResult();
 			}
 
-			var domainResult = application.Submit();
-
-			switch (domainResult)
+			var domainServiceResult = _applicationSubmissionService.SubmitApplication(application);
+			
+			switch (domainServiceResult)
 			{
 				case CommandValidationErrorResult:
-					return domainResult;
+					return domainServiceResult;
+				case CreateValidationErrorResult<IProject> createValidationErrorResult:
+					return domainServiceResult;
 				case CommandSuccessResult:
 					break;
+				case CreateSuccessResult<IProject> createSuccessResult:
+					await _projectCreateDataCommand.Execute(createSuccessResult.Payload);
+					break;
 				default:
 					throw new NotImplementedException("Other CreateResult types not expected");
 			}
+			
+			await _dataCommand.Execute(application);
 
-			if (application.ApplicationType == Domain.Core.ApplicationAggregate.ApplicationType.JoinAMat)
+			switch (domainServiceResult)
 			{
-				var projectResult = _projectFactory.Create(application);
-
-				switch (projectResult)
-				{
-					case CreateValidationErrorResult<IProject> createValidationErrorResult:
-						return new CommandValidationErrorResult(createValidationErrorResult.ValidationErrors);
-					case CreateSuccessResult<IProject> createSuccessResult:
-						await _projectCreateDataCommand.Execute(createSuccessResult.Payload);
-						break;
-					default:
-						throw new NotImplementedException("Other CreateResult types not expected");
-				}				
+				case CommandResult:
+					return domainServiceResult;
+				case CreateValidationErrorResult<IProject>:
+					// ToDo map this type
```

Comment:
What's the comment for?

Reply:
Well, it's because there is no mapper method in the CreateValidationErrorResult. Although there is no payload, there is still a generic type because the parent abstract class has a generic type. The CreateResult should probably have the mapper at this level. I was going to do it in a separate PR but I think I'll just do it here.